invisible_cities/reco/deconv_functions.py

```diff
@@ -0,0 +1,317 @@
+    """
+    Drops rogue/isolated hits (SiPMs) from a groupedby dataframe.
+
+    Parameters
+    ----------
+    df      : GroupBy ('event' and 'npeak') dataframe
+
+    Initialization parameters:
+        distance  : Distance to check for other sensors. Usually equal to sensor pitch.
+        variables : List with variables to be redistributed.
+
+    Returns
+    ----------
+    pass_df : hits after removing isolated hits
+    """
+    dist = np.sqrt(distance[0] ** 2 + distance[1] ** 2)
+
+    def drop_isolated_sensors(df : pd.DataFrame) -> pd.DataFrame:
+        x       = df.X.values
+        y       = df.Y.values
+        xy      = np.column_stack((x,y))
+        dr2     = cdist(xy, xy) # compute all square distances
+
+        if not np.any(dr2>0):
+            return df.iloc[:0] # Empty dataframe
+
+        closest = np.apply_along_axis(lambda d: d[d > 0].min(), 1, dr2) # find closest that it's not itself
+        mask_xy = closest <= dist # take those with at least one neighbour
+        pass_df = df.loc[mask_xy, :]
+
+        with np.errstate(divide='ignore'):
+            columns  = pass_df.loc[:, variables]
+            columns *= np.divide(df.loc[:, variables].sum().values, columns.sum())
+            pass_df.loc[:, variables] = columns
+
+        return pass_df
+
+    return drop_isolated_sensors
+
+
+def deconvolution_input(sample_width : List[float],
+                        bin_size     : List[float],
+                        inter_method : InterpolationMethod = InterpolationMethod.cubic
+                       ) -> Callable:
+    """
+    Prepares the given data for deconvolution. This involves interpolation of
+    the data.
+
+    Parameters
+    ----------
+    data        : Sensor (hits) position points.
+    weight      : Sensor charge for each point.
+
+    Initialization parameters:
+        sample_width : Sampling size of the sensors.
+        bin_size     : Size of the interpolated bins.
+        inter_method : Interpolation method.
+
+    Returns
+    ----------
+    Hs          : Charge input for deconvolution.
+    inter_points : Coordinates of the deconvolution input.
+    """
+
+    def deconvolution_input(data        : Tuple[np.ndarray, ...],
+                            weight      : np.ndarray
+                           ) -> Tuple[np.ndarray, Tuple[np.ndarray, ...]]:
+
+        ranges = [[coord.min() - 1.5 * sw, coord.max() + 1.5 * sw]     for coord,   sw in zip(data      , sample_width)]
+        nbin   = [np.ceil(np.diff(rang)/bs).astype('int')[0]           for bs   , rang in zip(bin_size  ,       ranges)]
+        if inter_method is not InterpolationMethod.none:
+            allbins = [np.linspace(*rang, np.ceil(np.diff(rang)/sw)+1) for rang ,   sw in zip(ranges[:2], sample_width)]
+            Hs, edges = np.histogramdd(data, bins=allbins, normed=False, weights=weight)
+        else:
+            Hs, edges = np.histogramdd(data, bins=nbin   , normed=False, weights=weight, range=ranges)
```

Comment:
Following the principle of not allowing values that are not a member of `InterpolationMethod`, I propose:
```
if inter_method in (InterpolationMethod.linear, InterpolationMethod.cubic):
    ...
elif inter_method is InterpolationMethod.none:
    ...
else:
    raise ValueError(...)
```